### ISSUE TITLE
How to implement a custom loss function if my ground truth value is a list?

### ISSUE THREAD

**Issue by sunshineatnoon** [fchollet/keras#1572]
I am solving a regression problem using neural network, I wrote a generator to generate batch data and use
```
model.fit_generator(generate_batch_data(vocPath,imageNameFile,100),nb_epoch=10,show_accuracy=True)
```
to train the model. In my `generate_batch_data function`, x is a 4D array contains a batch of images, but y is a list, each element in the list is an custom object containing the ground truth value for my problem, I want to use this list to calculate loss in my custom loss function. 

How can I implemented a custom loss function then? Can I assume that the parameter `y_pred` in `custom_objective(y_true, y_pred)` is just the list generated by my `generate_batch_data function` and I can treat it as a normal list? Thx!

**Comment by oeway** [comment]
Hi,
take a look at those loss functions in keras/keras/objectives.py

for example this one:
``` python
import numpy as np
from . import backend as K
def mean_squared_error(y_true, y_pred):
    return K.mean(K.square(y_pred - y_true), axis=-1)
```
You then can simply add an extra keyword `loss= custom_objective` in your model.compile.

In your case, I think y_true and y_pred is going to be a nD tensor, you should always use tensor operation on the last axis in your customized objective function.

**Comment by sunshineatnoon** [comment]
@oeway Thanks for your detailed reply, I will try this
